### DISCUSSION TITLE
Clarafying compresson levels

### DISCUSSION THREAD

**Discussion by ChaosNicro** [dpc/rdedup#220] [Q&A]
When initializing a repository, one can set a level for the given compression.
In the help it only states that they are normalized to 0 and can be adjusted from there. This likely helps with adjusting for different algorithms being slotted into the command structure, but I would like to know an effective range (eg. -5 to 9), as well as some indication of the mapping (eg. how to get the equivalent of 'zstd -15').

Is that documented somewhere?

**Comment by dpc** [comment]
IIRC it is being pass through to the underlying compression code, and they don't standardize or have hard limits in some cases. 

If you look at https://docs.rs/zstd/latest/src/zstd/lib.rs.html#39-42 , then at https://github.com/facebook/zstd/blob/909925785ae0c0f83f7a7d4738e5eb2a0cef78ad/build/single_file_libs/examples/roundtrip.c#L39 , it is 20 currently for zstd, but I think it used to be 22, and then 21, so :shrug:

**Reply by dpc** [reply]
BTW. IIRC it is totally fine to change the level after the repo was created (by editing some config file in there?). New chunks will be compressed using new compression level, while old ones will stay as is and keep being readable.
